hotshot-query-service/src/data_source/storage/sql.rs

```diff
@@ -810,6 +816,156 @@ impl VersionedDataSource for SqlStorage {
+
+                let leaf2_json =
+                    serde_json::to_value(leaf2.clone()).context("failed to serialize leaf2")?;
+                let qc2_json = serde_json::to_value(qc2).context("failed to serialize QC2")?;
+
+                // TODO (abdul): revisit after V1 VID has common field
+                let vid_common_bytes: Vec<u8> = row.try_get("vid_common")?;
+                let vid_share_bytes: Vec<u8> = row.try_get("vid_share")?;
+
+                let vid_share: ADVZShare = bincode::deserialize(&vid_share_bytes)
+                    .context("failed to serialize vid_share")?;
+
+                let new_vid_share_bytes = bincode::serialize(&VidShare::V0(vid_share))
+                    .context("failed to serialize vid_share")?;
+
+                vid_rows.push((leaf2.height() as i64, vid_common_bytes, new_vid_share_bytes));
+                leaf_rows.push((
+                    leaf2.height() as i64,
+                    commit.to_string(),
+                    leaf2.block_header().commit().to_string(),
+                    leaf2_json,
+                    qc2_json,
+                ));
+            }
+
+            // migrate leaf2
+            let mut query_builder: sqlx::QueryBuilder<Db> =
+                sqlx::QueryBuilder::new("INSERT INTO leaf2 (height, hash, block_hash, leaf, qc) ");
+
+            query_builder.push_values(leaf_rows.into_iter(), |mut b, row| {
+                b.push_bind(row.0)
+                    .push_bind(row.1)
+                    .push_bind(row.2)
+                    .push_bind(row.3)
+                    .push_bind(row.4);
+            });
+
+            let query = query_builder.build();
+
+            let mut tx = self.write().await.map_err(|err| QueryError::Error {
+                message: err.to_string(),
+            })?;
+
+            query.execute(tx.as_mut()).await?;
+
+            tx.commit().await?;
+            tracing::warn!("inserted {} rows into leaf2 table", offset);
+            // migrate vid
+            let mut query_builder: sqlx::QueryBuilder<Db> =
+                sqlx::QueryBuilder::new("INSERT INTO vid2 (height, common, share) ");
+
+            query_builder.push_values(vid_rows.into_iter(), |mut b, row| {
+                b.push_bind(row.0).push_bind(row.1).push_bind(row.2);
+            });
+
+            let query = query_builder.build();
+
+            let mut tx = self.write().await.map_err(|err| QueryError::Error {
+                message: err.to_string(),
+            })?;
+
+            query.execute(tx.as_mut()).await?;
+
+            tx.commit().await?;
+
+            tracing::warn!("inserted {} rows into vid2 table", offset);
+
+            if rows.len() < limit {
+                break;
+            }
+
+            offset += limit;
+        }
+
+        let mut tx = self.write().await.map_err(|err| QueryError::Error {
+            message: err.to_string(),
+        })?;
+
+        tracing::info!("query service types migration is completed!");
```

Comment:
Shouldn't we make this `warn` too?

Reply:
https://github.com/EspressoSystems/espresso-sequencer/pull/2478/commits/82155b510a1cf1a18531af54fada01fa0130344f